apps/web/src/pages/Applications/ApplicationSkillsPage/components/SkillForm.tsx

```diff
@@ -0,0 +1,316 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { useForm, FormProvider } from "react-hook-form";
+
+import {
+  Dialog,
+  Button,
+  Heading,
+  Accordion,
+  StandardAccordionHeader,
+  Well,
+} from "@gc-digital-talent/ui";
+import { Select, TextArea } from "@gc-digital-talent/forms";
+import { errorMessages } from "@gc-digital-talent/i18n";
+import { toast } from "@gc-digital-talent/toast";
+
+import { Experience, Scalars } from "~/api/generated";
+import {
+  deriveExperienceType,
+  getExperienceName,
+} from "~/utils/experienceUtils";
+import { useExperienceMutations } from "~/hooks/useExperienceMutations";
+
+const getSkillArgs = (
+  skillId: Scalars["ID"],
+  experience?: Experience,
+  details?: string,
+  remove?: boolean,
+) => {
+  const isExisting = experience?.skills?.find(
+    (experienceSkill) => experienceSkill.id === skillId,
+  );
+
+  if (!isExisting && !remove) {
+    return {
+      connect: [
+        {
+          id: skillId,
+          details,
+        },
+      ],
+    };
+  }
+
+  const newExperienceSkills = experience?.skills
+    ?.filter((experienceSkill) =>
+      remove ? experienceSkill.id !== skillId : true,
+    )
+    ?.map(({ id, experienceSkillRecord }) => ({
+      id,
+      details: id !== skillId ? experienceSkillRecord?.details : details,
+    }));
+
+  return {
+    sync: newExperienceSkills,
```

Comment:
A bit complicated way to do this. 
Wouldn't adding `disconnect` to the schema simplify this?

![image](https://user-images.githubusercontent.com/40485260/236534608-89a6d272-edfc-4923-9679-8e153fdee8cb.png)

Reply:
Yeah, I wasn't sure if there was a reason we did not have a `disconnect` but that is a good idea. Added in [update to use disconnect](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6518/commits/1f2153dbb2df8ae97c7fa6bbffb71368a64b5c21)

Reply:
Much nicer!